src/core/unif.mli

```diff
@@ -0,0 +1,9 @@
+open Term
+
+(** [solve_noexn ~type_check p] tries to simplify the constraints of [p]. It
+   returns [false] if it finds a constraint that cannot be
+   satisfied. Otherwise, [p.to_solve] is empty but [p.unsolved] may still
+   contain constraints that could not be simplified. Metavariable
+   instantiations are type-checked only if the optional argument [~type_check]
+   is [true] (default). *)
```

Comment:
I'm picky, but I think just writing `(default)` might be confusing when reading the comment for the first time, I'd be more explicit to say that the argument `~type_check` defaults to `true`.